dali/kernels/signal/fft/fft_postprocess_test.cu

```diff
@@ -0,0 +1,221 @@
+  power_dB f;
+  EXPECT_NEAR(f(float2{0,0}), -80, 1e-5f);
+  EXPECT_EQ(f(float2{1,0}), 0);
+  EXPECT_EQ(f(float2{0,-1}), 0);
+
+  EXPECT_NEAR(f(float2{1,1}), 3.01029995664, 1e-5f);
+  EXPECT_NEAR(f(float2{2,0}), 6.02059991328, 1e-5f);
+  EXPECT_NEAR(f(float2{2,2}), 9.03089986992, 1e-5f);
+}
+
+template <typename Out, typename In, typename Convert = identity>
+struct FFTPostprocessArgs {};
+
+template <typename TestArgs>
+class FFTPostprocessTest;
+
+template <typename Out, typename In, typename Convert>
+class FFTPostprocessTest<FFTPostprocessArgs<Out, In, Convert>> : public ::testing::Test {
+ public:
+  void ToFreqMajor() {
+    std::mt19937_64 rng;
+    TensorListShape<2> in_shape, out_shape;
+    std::uniform_int_distribution<int> dist(1, 500);
+    int N = 10;
+    int fft = 200;  // deliberately not a multiple of 32
+    in_shape.resize(N);
+    out_shape.resize(N);
+    for (int i = 0; i < N; i++) {
+      int len = dist(rng);
+      in_shape.set_tensor_shape(i, { len, fft });
+      out_shape.set_tensor_shape(i, { fft, len });
+    }
+    TestTensorList<In, 2> in;
+    TestTensorList<Out, 2> out, ref;
+    in.reshape(in_shape);
+    auto cpu_in = in.cpu();
+    UniformRandomFill(cpu_in, rng, -1, 1);
+
+    ToFreqMajorSpectrum<Out, In, Convert> tr;
+    KernelContext ctx;
+    ScratchpadAllocator sa;
+    KernelRequirements req = tr.Setup(ctx, in_shape);
+    ASSERT_EQ(req.output_shapes.size(), 1u);
+    ASSERT_EQ(req.output_shapes[0], out_shape);
+    sa.Reserve(req.scratch_sizes);
+    auto scratchpad = sa.GetScratchpad();
+    ctx.scratchpad = &scratchpad;
+    out.reshape(out_shape);
+    tr.Run(ctx, out.gpu(), in.gpu());
+    CUDA_CALL(cudaGetLastError());
+
+    auto cpu_out = out.cpu();
+    ref.reshape(out_shape);
+    auto cpu_ref = ref.cpu();
+
+    Convert convert;
+    for (int i = 0; i < N; i++) {
+      TensorView<StorageCPU, In, 2> in_tv = cpu_in[i];
+      TensorView<StorageCPU, Out, 2> ref_tv = cpu_ref[i];
+      for (int y = 0; y < in_tv.shape[0]; y++)
+        for (int x = 0; x < in_tv.shape[1]; x++)
+          *ref_tv(x, y) = convert(*in_tv(y, x));
+    }
+
+    double eps = std::is_same<Convert, identity>::value ? 0 : 1e-5;
+
+    Check(cpu_out, cpu_ref, EqualEps(eps));
+  }
+
+  void ConvertTimeMajorInPlace() {
+    std::mt19937_64 rng;
+    TensorListShape<2> in_shape, out_shape;
+    std::uniform_int_distribution<int> dist(1, 500);
+    int N = 10;
+    int fft = 200;  // deliberately not a multiple of 32
+    in_shape.resize(N);
+    out_shape.resize(N);
+    for (int i = 0; i < N; i++) {
+      int len = dist(rng);
+      int ratio = sizeof(In) / sizeof(Out);
+      in_shape.set_tensor_shape(i, { len, fft });
+      out_shape.set_tensor_shape(i, { len, fft * ratio });
+    }
+    TestTensorList<In, 2> in;
+    TestTensorList<Out, 2> out, ref;
+    TensorListView<StorageGPU, Out, 2> out_gpu;
+    in.reshape(in_shape);
+    auto cpu_in = in.cpu();
+    UniformRandomFill(cpu_in, rng, -1, 1);
```

Comment:
Seems to be very similar to `ToFreqMajor`, can you extract this to a common fucntion?

Reply:
I'll try, but the code here consists mostly of declarations. I could make some of these members, but I'm not quite sure about the lifetime of the test object.

Reply:
Suggestion, not a must.